framework/web/Response.php

```diff
@@ -488,6 +485,9 @@ public function sendContentAsFile($content, $attachmentName, $options = [])
             $this->content = $content;
         }
 
+        $mimeType = isset($options['mimeType']) ? $options['mimeType'] : 'application/octet-stream';
+        $this->setDownloadHeaders($attachmentName, $mimeType, !empty($options['inline']), $end - $begin + 1);
```

Comment:
I disagree because `$this->content` always has `$end - $begin + 1` length. It's guaranteed by both `$this->getHttpRange` and `StringHelper::byteSubstr` functions.